rust/datafusion/src/execution/context.rs

```diff
@@ -455,12 +495,11 @@ impl ExecutionContext {
         input_schema: &Schema,
     ) -> Result<Arc<dyn PhysicalExpr>> {
         match e {
-            Expr::Alias(expr, name) => {
-                let expr = self.create_physical_expr(expr, input_schema)?;
-                Ok(Arc::new(Alias::new(expr, &name)))
-            }
-            Expr::Column(i) => {
-                Ok(Arc::new(Column::new(*i, &input_schema.field(*i).name())))
+            Expr::Alias(expr, ..) => Ok(self.create_physical_expr(expr, input_schema)?),
```

Comment:
It looks like we're losing the alias name here?

Reply:
So we just need to fix this so that the physical expression is created using the aliased name.

Reply:
No, for the same reason: the physical plan does not care about names anymore, and aliases are only about naming.

Reply:
fyi my comment was written before I read your second comment!

Reply:
Ok, so the schema for these physical operators is based on the logical schema? I guess I just want to be sure that the alias name is preserved in the query results.

A unit test demonstrating this would be good.

Reply:
(I wrote my other comment because I became uncertain about my claim and wanted to double check ^_^: yes, good to have this triple checked; this is a big change)

Let's see:

1. the schema of the logical plan is built on the column's names as per [this line](https://github.com/apache/arrow/pull/7796/files#diff-be9adb93b0effc41a9672892e1aed3e1R1012), which is then passed to the physical plan as per line of this discussion. So, it should be the case...
2. [this test](https://github.com/apache/arrow/pull/7796/files#diff-8273e76b6910baa123f3a25a967af3b5R1089) IMO demonstrates this: we alias an expression in the logical plan, and the physical's plan schema's field of that column is the alias itself.

Reply:
Thanks. I just pulled your branch and modified one of the integration tests just to be sure, and it looks good. This really helps simplify some things, thanks!

Could you rebase?

Reply:
:) perfect. I have rebased this against the latest master.